### PR TITLE
Disable wait_pop_timed timer in unbounded case to reduce log spam.

### PR DESCRIPTION
**Pull request by jvilk-stripe**
Disable wait_pop_timed timer in unbounded case to reduce log spam.

### Motivation

Silences logspam that happens every 250ms from typechecker coord thread. Unlike bounded usages of wait_pop_timed, the typechecker coord thread uses a unbounded queue that sets the expected number of queue entries to be MAX INT to create an effectively unbounded queue.

### Test plan

N/A

### CODE REVIEW

**Comment by jvilk-stripe** [comment]
cc @mkillianey-stripe who noticed this issue (thanks!)

File: common/concurrency/ConcurrentQueue.h [hunk 1]
```diff
@@ -60,7 +62,12 @@ template <class Elem, class Queue> class AbstractConcurrentBoundedQueue {
         if (!sorbet::emscripten_build) {
             ret.shouldRetry = elementsLeftToPush.load(std::memory_order_acquire) != 0;
             if (ret.shouldRetry) {
-                sorbet::Timer time(log, "wait_pop_timed");
+                std::unique_ptr<sorbet::Timer> time;
```
Comment (DarkDimius) [inline common/concurrency/ConcurrentQueue.h]: WDYT about removing the timer from here instead?
The reasoning for keeping it here was that emitting a log line every ~250ms is fine.

Now, I think it's a surprising implication that using unbounded collection removes timers.
WDYT?

**Comment by jvilk-stripe** [comment]
@DarkDimius The problem is that, while sorbet is in LSP mode, it continuously emits a log line for wait_pop_timed every 250 ms. I wanted to prevent that from happening.

I see value in the log line for bounded cases for traces, so I figured it was important to keep in for that use case.

Would it be better to pass a `silent` arg to wait_pop_timed that silences the timer?

**Comment by DarkDimius** [comment]
> Would it be better to pass a silent arg to wait_pop_timed that silences the timer?

sgtm

**Comment by jvilk-stripe** [comment]
Made adjustments; waiting for build to pass.

**Pull request merged**